source/Nuke.Common/Execution/DockerExecutor.cs

```diff
@@ -0,0 +1,166 @@
+
+            CopyPackageReferences();
+
+            var workingDirectory = attribute.Platform == PlatformFamily.Windows ? "c:\\Build" : "/build";
+
+            var path = attribute.Platform == PlatformFamily.Windows
+                ? $"c:\\build\\{NukeBuild.RootDirectory.GetWinRelativePathTo(NukeBuild.TemporaryDirectory)}\\nukebuild\\{Path.GetFileName(NukeBuild.BuildAssembly)}"
+                : $"/build/{NukeBuild.RootDirectory.GetUnixRelativePathTo(NukeBuild.TemporaryDirectory)}/nukebuild/{Path.GetFileName(NukeBuild.BuildAssembly)}";
+
+            var args = new[]
+                       {
+                           path,
+                           "--nologo",
+                           "--skip",
+                           "--host", "NukeInDocker",
+                           "--target", $"\"{target.Name}\"",
+                           "--root", workingDirectory
+                       };
+
+            var userProfile = EnvironmentInfo.SpecialFolder(SpecialFolders.UserProfile);
+            var tempFile = GetEnvFile(workingDirectory, userProfile);
+
+            try
+            {
+                var volumes = new []
+                              {
+                                  $"{NukeBuild.RootDirectory}:{workingDirectory}"//,
+                                  //$"{userProfile}:{userProfile}"
+                              };
+                DockerTasks.DockerRun(settings => settings
+                    .EnableRm()
+                    .SetImage(attribute.Image)
+                    .SetVolume(volumes)
+                    .SetCommand("dotnet")
+                    .SetWorkdir(workingDirectory)
+                    .SetEnvFile(tempFile)
+                    .SetArgs(args.Concat(attribute.Args)));
+            }
+            finally
+            {
+                //File.Delete(tempFile);
+            }
+        }
+
+        private static void CopyPackageReferences()
+        {
+            Log.Information("Inlining PackageDownload references from build project");
+            var buildProject = ProjectModelTasks.ParseProject(NukeBuild.BuildProjectFile).NotNull();
+            foreach (var item in buildProject.Items)
+            {
+                if (item.ItemType == "PackageDownload")
+                {
+                    var version = item.Metadata.First(x => x.Name == "Version").EvaluatedValue.Replace("[", "").Replace("]", "");
+                    var packageId = item.EvaluatedInclude;
+                    Log.Information("Inlining {PackageId} {Version} reference from build project", packageId, version);
+                    var package = NuGetPackageResolver.GetLocalInstalledPackage(packageId, NukeBuild.BuildProjectFile, version);
+                    FileSystemTasks.CopyDirectoryRecursively(package.Directory, NukeBuild.TemporaryDirectory / "nukebuild" / packageId);
+                }
+            }
+        }
+
+        private static void PublishBuildProject(RunInDockerContainerAttribute attribute)
```

Comment:
Need to publish a copy of the project, as we cant re-run the `build.sh`/`build.ps1` files as they try to recompile the build project, hitting locked files.